batch_seqr_loader/seqr_load.py

```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env python3
+
+"""
+Hail script to submit on a dataproc cluster. Converts input multi-sample VCFs
+into a matrix table, which annotates and prepares to load into ES.
+"""
+
+import logging
+from typing import Optional, List
+
+import click
+import hail as hl
+
+from luigi_pipeline.lib.model.seqr_mt_schema import SeqrVariantsAndGenotypesSchema
+from hail_scripts.v02.utils.elasticsearch_client import ElasticsearchClient
+
+logger = logging.getLogger()
+
+
+@click.command()
+@click.option(
+    '--source-path',
+    'source_paths',
+    multiple=True,
+    required=True,
+)
+@click.option(
+    '--dest-mt-path',
+    'dest_path',
+    required=True,
+)
+@click.option(
+    '--bucket',
+    'bucket',
+    required=True,
+)
+@click.option('--genome-version', 'genome_version', default='GRCh38')
+@click.option(
+    '--reference-ht',
+    'reference_path',
+    required=True,
+    help='Path to the Hail table storing the reference variants.',
+    default='gs://seqr-reference-data/GRCh38/all_reference_data/v2/combined_reference_data_grch38-2.0.2.ht',
+)
+@click.option(
+    '--clinvar-ht',
+    'clinvar_path',
+    required=True,
+    help='Path to the Hail table storing the clinvar variants.',
+    default='gs://seqr-reference-data/GRCh38/clinvar/clinvar.GRCh38.ht',
+)
+@click.option(
+    '--hgmd-ht',
+    'hgmd_path',
+    help='Path to the Hail table storing the hgmd variants.',
+    # default='gs://seqr-reference-data-private/GRCh38/HGMD/hgmd_pro_2018.4_hg38_without_db_field.vds'
+)
+@click.option('--disable-validation', 'disable_validation', is_flag=True)
+@click.option(
+    '--sample-type', 'sample_type', type=click.Choice(['WGS', 'WES']), default='WGS'
+)
+@click.option(
+    '--dataset-type',
+    'dataset_type',
+    type=click.Choice(['VARIANTS', 'SV']),
+    default='VARIANTS',
+)
+@click.option(
+    '--remap-tsv',
+    'remap_path',
+    help='Path to a TSV file with two columns: s and seqr_id.',
+)
+@click.option(
+    '--subset-tsv',
+    'subset_path',
+    help='Path to a TSV file with one column of sample IDs: s.',
+)
+@click.option('--vep-block-size', 'vep_block_size')
+def main(
+    source_paths: List[str],
+    dest_path: str,
+    bucket: str,
+    genome_version: str,
+    reference_path: str,
+    clinvar_path: str,
+    hgmd_path: Optional[str],
+    disable_validation: bool,
+    sample_type: str,
+    dataset_type: str,  # pylint: disable=unused-argument
+    remap_path: str = None,  # pylint: disable=unused-argument
+    subset_path: str = None,  # pylint: disable=unused-argument
+    vep_block_size: Optional[int] = None,
+):  # pylint: disable=missing-function-docstring
+    print('Running')
+    genome_version = genome_version.replace('GRCh', '')
+    mt = import_vcf(source_paths, genome_version)
+    mt = annotate_old_and_split_multi_hts(mt)
+    mt.write(f'{bucket}/annotate_old_and_split_multi_hts.mt', overwrite=True)
```

Comment:
Are these debug writes? If they're checkpointing, they probably need to be better scoped to an output-directory rather than just the bucket.

Reply:
Dropped this!